lib/pbench/cli/server/user_management.py

```diff
@@ -0,0 +1,225 @@
+def user_command_cli(ctx):
+    pass  # Entry Point
+
+
+@user_command_cli.command()
+@common_options
+@click.option(
+    "--username",
+    prompt=True,
+    required=True,
+    help="pbench server account username (will prompt if unspecified)",
+)
+@click.option(
+    "--password",
+    prompt=True,
+    hide_input=True,
+    required=True,
+    help="pbench server account password (will prompt if unspecified)",
+)
+@click.option(
+    "--email",
+    prompt=True,
+    required=True,
+    help="pbench server account email (will prompt if unspecified)",
+)
+@click.option(
+    "--first-name",
+    prompt=True,
+    required=True,
+    help="pbench server account first name (will prompt if unspecified)",
+)
+@click.option(
+    "--last-name",
+    prompt=True,
+    required=True,
+    help="pbench server account last name (will prompt if unspecified)",
+)
+@click.option(
+    "--role",
+    type=click.Choice([role.name for role in Roles], case_sensitive=False),
+    required=False,
+    help="Optional role of the user such as Admin",
+)
+@pass_cli_context
+def user_create(
+    context: object,
+    username: str,
+    password: str,
+    email: str,
+    first_name: str,
+    last_name: str,
+    role: str,
+) -> None:
+    try:
+        # Setup the pbench server config and db access
+        config_setup(context, "pbench-user-create")
+
+        user = User(
+            username=username,
+            password=password,
+            first_name=first_name,
+            last_name=last_name,
+            email=email,
+            role=role if role else "",
+        )
+        user.add()
+        if user.is_admin():
+            click.echo(f"Admin user {username} registered")
+        else:
+            click.echo(f"User {username} registered")
+        rv = 0
+    except Exception as exc:
+        click.echo(exc, err=True)
+        rv = 2 if isinstance(exc, BadConfig) else 1
+
+    click.get_current_context().exit(rv)
+
+
+# User delete CLI
+@user_command_cli.command()
+@common_options
+@click.argument("username")
+@pass_cli_context
+def user_delete(context: object, username: str) -> None:
+    try:
+        # Setup the pbench server config and db access
+        config_setup(context, "pbench-user-delete")
+
+        # Delete the the user with specified username
+        if User.delete(username=username):
+            click.echo(f"User {username} deleted")
+            rv = 0
+        else:
+            click.echo(f"User {username} does not exists")
+            rv = 1
+    except Exception as exc:
+        click.echo(exc, err=True)
+        rv = 2 if isinstance(exc, BadConfig) else 1
+
+    click.get_current_context().exit(rv)
+
+
+# Users list CLI
+@user_command_cli.command()
+@common_options
+@pass_cli_context
+def user_list(context: object) -> None:
+    try:
+        # Setup the pbench server config and db access
+        config_setup(context, "pbench-user-list")
+
+        ROW_FORMAT = "{0:15}\t{1:15}\t{2:15}\t{3:15}\t{4:20}"
```

Comment:
I recommend that you move this declaration to module scope so that you can import it and use it in the unit test.